app/routes/routes.py

```diff
@@ -0,0 +1,107 @@
+from app import db
+from app.models.planet import Planet
+from flask import Blueprint, jsonify, abort, make_response, request
+from .routes_helper import error_message
+
+planets_bp = Blueprint("planets_bp", __name__, url_prefix="/planets")
+
+def make_planet_safely(data_dict):
+    try:
+        return Planet.from_dict(data_dict)
+    except KeyError as err:
+        error_message(f"Missing key: {err}", 400)
+
+def replace_planet_safely(planet, data_dict):
+    try:
+        planet.replace_details(data_dict)
+    except KeyError as err:
+        error_message(f"Missing key: {err}", 400)
+
+def get_planet_record_by_id(id):
+    try:
+        id = int(id)
+    except ValueError:
+        error_message(f"Invalid id {id}", 400)
+
+    planet = Planet.query.get(id)
+
+    if planet:
+        return planet
+    
+    error_message(f"No planet with id {id} found", 404)
+
+# GET /planets
+@planets_bp.route("", methods = ["GET"])
+def read_all_planets():
+    name_param = request.args.get("name")
+
+    if name_param:
+        planets = Planet.query.filter_by(name=name_param.capitalize())
+        
+    else:
+        planets = Planet.query.all()
+
+    result_list = [planet.to_dict() for planet in planets]
+
+    if not result_list:
+        return jsonify("No planets found with that name.")
```

Comment:
Hmm, what happens if you try to get all planets, but there are no planets stored in the db yet? It looks like `result_list` would be an empty list after line 44, so we'd enter the if-block on line 46 and print the error `"No planets found with that name."` even though it doesn't quite apply. How could we make sure that message is only returned when relevant?